src/cpp/shared/scripting/scripting.cpp

```diff
@@ -47,10 +47,10 @@ void MakeLuaHeroTable(lua_State *L, void *ptrAddr) {
   lua_setmetatable(L,-2);
 }
 
-void* GetPointerFromLuaClassTable(lua_State *L, int argNum) {
+void* GetPointerFromLuaClassTable(lua_State *L, int argNumber, int numArgs) {
   lua_pushstring(L, "ptr");
-  lua_gettable(L, -2);
-  void* ret = (void*)(int)lua_tonumber(L, -argNum);
+  lua_gettable(L, -numArgs-1);
+  void* ret = (void*)(int)lua_tonumber(L, -argNumber);
```

Comment:
Hmmm. It seems to me that you want lua_tonumber to run on the value which had been returned form gettable, which lies on top of the stack, and so  this only works if -argNumber == -1, i.e.: if argNumber == 1. 

And, indeed, at every caller, you do have argNumber = 1.  So why have this as a parameter?